apps/remote_control/lib/lexical/remote_control/build/error.ex

```diff
@@ -49,21 +49,36 @@ defmodule Lexical.RemoteControl.Build.Error do
     )
   end
 
-  defp uniq(diagnostics) do
+  def uniq(diagnostics) do
     # We need to uniq by position because the same position can be reported
     # and the `end_line_diagnostic` is always the precise one
     extract_line = fn
       %Result{position: {line, _column}} -> line
       %Result{position: {start_line, _start_col, _end_line, _end_col}} -> start_line
-      %Result{position: line} when is_integer(line) -> line
+      %Result{position: line} -> line
     end
 
-    Enum.uniq_by(diagnostics, extract_line)
+    # Note: Sometimes error and warning appear on one line at the same time
+    # So we need to uniq by line and severity,
+    # and :error is always more important than :warning
+    extract_line_and_severity = &{extract_line.(&1), &1.severity}
```

Comment:
but what I really want is to be sorted by line and severity, and unique by line.

for example: 

```elixir
    test "reports multiple errors", %{project: project} do
      source = ~S[
        defmodule WithFiveErrors do
          def error(unused_1, unused_2) do
            unknown_fn()
            unused_3 = other_unknown()
          end
        end
      ]
```

if we only use `line_and_severity`, it will return a duplicated error in one line, which is noise. 

```elixir
    [
      %Lexical.Plugin.V1.Diagnostic.Result{
        details: nil,
        message: "undefined function
 other_unknown/0",
        position: 5,
        severity: :error,
        source: "Elixir",
        uri:
          "file:///Users/scottming/Code/lexical/apps/remote_control/test/fixtures/project_metadata/3/file.exs"
      },
      %Lexical.Plugin.V1.Diagnostic.Result{
        details: nil,
        message: "variable \"unused_3\" is unused",
        position: {5, 13},
        severity: :warning,
        source: "Elixir",
        uri:
          "file:///Users/scottming/Code/lexical/apps/remote_control/test/fixtures/project_metadata/3/file.exs"
      }
    ]
```

Having the editor show one diagnostics on a line is enough, and the user will naturally see the other one after the fix.

If multiple are on one line, then there is really a lot of noise, like in this example before 1.15.

![image](https://github.com/lexical-lsp/lexical/assets/12830256/1295ec2e-ce4d-413f-9359-ee812f4941db)